controllers/drplacementcontrol_controller.go

```diff
@@ -760,16 +760,39 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		return fmt.Errorf("failed to get DRPolicy while finalizing DRPC (%w)", err)
 	}
 
-	// delete manifestworks (VRG)
+	// delete manifestworks (VRGs)
 	for _, drClusterName := range rmnutil.DrpolicyClusterNames(drPolicy) {
 		err := mwu.DeleteManifestWorksForCluster(drClusterName)
 		if err != nil {
 			return fmt.Errorf("%w", err)
 		}
+	}
+
+	// delete cloned placementrule, if created
+	if preferredCluster == "" {
+		return r.deleteClonedPlacementRule(ctx, clonedPlRuleName, drpc.Namespace)
+	}
+
+	// Verify VRGs have been deleted
+	vrgs, err := r.getVRGsFromManagedClusters(drpc, drPolicy)
```

Comment:
If last action was Failover and the preferredCluster is down at the moment of deletion, we will not wait for it to come back and then delete. This is because in getVRGsFromManagedClusters ignores errors from a down cluster if it was a cluster that we are not failing over to.

Now, question is should we wait?

Reply:
I don't think we should wait.  Who knows if the failed cluster will ever be coming back.  The cleanup that we added in this PR is "best-effort".

Reply:
(not for now) We may need to wait (IOW, leave the DRPC resource un-deleted), so that user can take force options (like removing the finalizer, which they need to understand the implications of). If we remove the DRPC, the user has no way of knowing if resources on the actual clusters are leaked and what (if possible) needs to be cleaned up.

I would also think a user should remove a managed cluster from the hub if it is unrechable and gone for good, at which point "maybe" we can also safely delete DRPC and forget about the unrechable cluster.

But, this is not for this PR. Something to look at and address as appropriate later.